concepts/functions/about.md

```diff
@@ -1,3 +1,302 @@
 # About
 
-TODO: add information on functions concept
+A [`function`][function] is a block of organized, reusable code that is used to perform a single, related action.
+`Functions` provide better modularity for your application and a high degree of code reusing.
+Like other programming languages, python provides _pre-defined functions_ ([`print`][print], [`map`][map] etc) that are readily available in the Python language. 
+However, you can define your own functions as well which are called [`user-defined functions`][user defined functions].
+Functions can be as simple as _printing a message to the console_ or as complex as _performing a calculation_.
+
+To execute the code inside a function, you need to call the function, which is done by using the function name followed by parenthesis [`()`].
+Data known as `parameters` can be passed to the function inside the parenthesis. Broader terms for parameters are `arguments`. Functions can perform different tasks depending on the value of the parameters.
+
+The function can also return a value using the `return` keyword. The value returned by the function is known as the `return value`. The return value is returned to the caller of the function.
+
+## Creation
+
+In python, functions are created using the `def` keyword. The function definition is followed by the function name and parenthesis [`()`]. Inside the parenthesis, the parameters are specified, and are separated by commas. After the parameters, the colon (`:`) is used to separate the function header from the function body.
+
+The function body is a block of code that is executed when the function is called. The body of the function is indented. The indentation is important because the function body is a block of code. A value can be returned from the function by using the `return` keyword, which will be used by the caller of the function.
```

Comment:
```suggestion
The function body is a block of code that is executed when the function is called. The body of the function is indented. The indentation is important because Python relies on it to know where that block of code ends. A value can be returned from the function by using the `return` keyword, which can then be used by the caller of the function.
```